### DISCUSSION TITLE
How to merge Spatial data

### DISCUSSION THREAD

**Discussion by EZUY** [satijalab/seurat#7096] [Q&A] [answered]
Hi All,
I'm currently trying to merge multiple spatial data generated with spaceranger count.

I first tried to use aggregated matrix with spaceranger aggr
```
data_dir<-"Seurat\\Aggr"
A1_10X_Spatial<-Load10X_Spatial(data.dir = data_dir,filename ="filtered_feature_bc_matrix.h5")`
```
Then I have 
_Error in readPNG(source = file.path(image.dir, image.name)) : 
unable to open Seurat\Aggr/spatial/tissue_lowres_image.png_

```
a1 <- Load10X_Spatial(data.dir="Seurat\\A1", filename = "filtered_feature_bc_matrix.h5", assay = "Spatial", slice="slice1")
b1 <- Load10X_Spatial(data.dir="Seurat\\B1", filename = "filtered_feature_bc_matrix.h5", assay = "Spatial", slice="slice2")
c1 <- Load10X_Spatial(data.dir="Seurat\\C1", filename = "filtered_feature_bc_matrix.h5", assay = "Spatial", slice="slice3")
d1 <- Load10X_Spatial(data.dir="Seurat\\D1", filename = "filtered_feature_bc_matrix.h5", assay = "Spatial", slice="slice4")
a1 <- SCTransform(a1, assay = "Spatial", verbose = FALSE)
b1 <- SCTransform(b1, assay = "Spatial", verbose = FALSE)
c1 <- SCTransform(c1, assay = "Spatial", verbose = FALSE)
d1 <- SCTransform(d1, assay = "Spatial", verbose = FALSE)
a1[["orig.ident"]] <- "a1"
b1[["orig.ident"]] <- "b1"
c1[["orig.ident"]] <- "c1"
d1[["orig.ident"]] <- "d1"
st_seurat_merge <- a1
st_seurat_merge <- merge(st_seurat_merge, a1)
st_seurat_merge <- merge(st_seurat_merge, b1)
st_seurat_merge <- merge(st_seurat_merge, c1)
st_seurat_merge <- merge(st_seurat_merge, d1)

```
when I'm merging them. I got _Warning: Some cell names are duplicated across objects provided. Renaming to enforce unique cell names.Warning: Some cell names are duplicated across objects provided. Renaming to enforce unique cell names._

when I run 
```
st_seurat_merge <- RunPCA(st_seurat_merge, assay = "SCT", verbose = FALSE)
st_seurat_merge <- FindNeighbors(st_seurat_merge, reduction = "pca", dims = 1:30)
st_seurat_merge <- FindClusters(st_seurat_merge, verbose = FALSE)
st_seurat_merge <- RunUMAP(st_seurat_merge, reduction = "pca", dims = 1:30)
```
I got 
_Error in PrepDR(object = object, features = features, verbose = verbose) : 
Variable features haven't been set. Run FindVariableFeatures() or provide a vector of feature names._

Usually, how do I merge different spatial matrix&picture to one Seurat object?

**Comment by EZUY** [comment, marked as answer]
nvm. I added 

```
DefaultAssay(st_seurat_merge) <- "SCT"
VariableFeatures(st_seurat_merge) <- c(VariableFeatures(a1), VariableFeatures(b1),VariableFeatures(c1), VariableFeatures(d1))
```

**Comment by EZUY** [comment]
May I know how to get the gene expression in different clusters of merged Spatial data?

**Reply by Pancreas-Pratik** [reply]
Hi @EZUY

I have two VisiumHD datasets (same tissue, different conditions) running through `spaceranger count` at the moment, but am wondering how to appropriately proceed next.

What did you end up doing?

Did you use the output from `spaceranger count` only? or did you use any ouputs from `spaceranger aggr`?

and then what did you do afterwards for downstream analysis? was merging necessary?

Thank you very much in advance.

-----

EDIT: I think I found the solution that I will try (which looks like what you did), which leads me to believe that maybe `spaceranger aggr` output can be used in some way:
https://satijalab.org/seurat/articles/spatial_vignette#working-with-multiple-slices-in-seurat

and also try with a combination of this section for DGE analysis:
https://satijalab.org/seurat/articles/visiumhd_analysis_vignette#unsupervised-clustering

-----

EDIT: spaceranger v3.1.1 does not support `aggr` for VisiumHD
